libs/blocks/modal/modal.js

```diff
@@ -246,6 +249,7 @@ export function delayedModal(el) {
 export default function init(el) {
   const { modalHash } = el.dataset;
   if (delayedModal(el) || window.location.hash !== modalHash || document.querySelector(`div.dialog-modal${modalHash}`)) return null;
+  if (dialogLoadingSet.has(modalHash?.replace('#', ''))) return null;
```

Comment:
I'd suggest either a comment, unit test or both. Since it's not super obvious on why this line of code exists
```suggestion
  if (dialogLoadingSet.has(modalHash?.replace('#', ''))) return null; // prevent duplicate modal loading
```

Reply:
Thanks @mokimo ! I personally feel that a comment would be enough, because otherwise it will bloat up modal's unit tests by quite some boilerplate stuff given how modal.test.js is currently structured and we'll need to mock the fetch again. Just from the code level, this change should be trivial enough for us to just feel confident 😆